core/src/test/scala/unit/kafka/admin/TopicCommandTest.scala

```diff
@@ -592,4 +594,32 @@ class TopicCommandTest extends ZooKeeperTestHarness with Logging with RackAwareT
     expectAlterInternalTopicPartitionCountFailed(Topic.GROUP_METADATA_TOPIC_NAME)
     expectAlterInternalTopicPartitionCountFailed(Topic.TRANSACTION_STATE_TOPIC_NAME)
   }
+
+  @Test
+  def testDescribeTopicsWithNoLeaders(): Unit = {
+    TestUtils.createBrokersInZk(zkClient, List(0, 1, 2))
+    val topic = "test-topic"
+
+    // create the topics
+    val createOpts = new TopicCommandOptions(Array("--partitions", "1", "--replication-factor", "2", "--topic", topic))
+    topicService.createTopic(createOpts)
+    val leaderIsrAndControllerEpochs = Map(
+      new TopicPartition(topic, 0) -> LeaderIsrAndControllerEpoch(
+        LeaderAndIsr(leader = 0, leaderEpoch = 0, isr = List(0, 1), zkVersion = 0),
+        controllerEpoch = 0)
+    )
+    topicService.zkClient.createTopicPartitionStatesRaw(leaderIsrAndControllerEpochs, ZkVersion.MatchAnyVersion)
```

Comment:
Rather than manually updating the leader and isr state, which may have unexpected behavior, wouldn't it be easier to set the replica assignment in the create topic command?